.github/workflows/codegen.yml

```diff
@@ -8,18 +8,80 @@ name: codegen
 jobs:
   discovery:
     uses: googleapis/discovery-artifact-manager/.github/workflows/list-services.yml@master
-
-  generate:
+  batch:
     runs-on: ubuntu-20.04
     needs: discovery
+    outputs:
+      services: ${{ steps.chunk.outputs.result }}
+    steps:
+      - uses: actions/github-script@v5
+        id: chunk
+        with:
+          script: |
+            console.log('splitting service names list into batches')
+            const services = ${{ needs.discovery.outputs.services }}
+            const result = {
+              "one": [],
+              "two": [],
+            };
+            for (let i = 0; i < services.length; i++) {
+              if (services[i] != "discovery" && services[i] != "websecurityscanner") {
+                j = i % 2 ? "one" : "two";
+                result[j].push(services[i])
+              }
+            }
+            return result
+  generate_batch1:
+    runs-on: ubuntu-20.04
+    needs: batch
+    strategy:
+      fail-fast: false
```

Comment:
Why is this false?
A true value will help fail the script of the codegen for any service fails. 
Its also false for `generate_batch2`

Reply:
If we have an error in one service, we want to keep generating the other services. We don't want one service to hold up service updates for every other service.

An example of when this has happened - APIs occasionally publish two messages which names only vary by casing (see https://github.com/googleapis/google-api-php-client-services/blob/main/.gitignore#L4). This breaks installs on MacOS because it's a case-insensitive filesystem. But other APIs were still updated while this update failed.